File: python/1.py
# Two Sum
# 两数之和


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        comp_dict = {}
        for i, num in enumerate(nums):
            comp = target - num
            if comp in comp_dict:
                return [comp_dict[comp], i]
            comp_dict[num] = i
        return []

if __name__ == '__main__':
    test_cases = [
        ([2, 7, 11, 15], 9),
        ([3, 2, 4], 6),
        ([3, 3], 6),
    ]

    sol = Solution()
    for case in test_cases:
        res = sol.twoSum(*case)
        print(case, '\n', res)

File: cpp/1.cpp
#include "common.h"

class Solution
{
public:
    std::vector<int> twoSum(std::vector<int> &nums, int target)
    {
        int len = nums.size();
        std::vector<int> idxs = {0, 1};
        while (true) {
            for(idxs[1] = idxs[0]+1; idxs[1] < len; idxs[1]++) {
                if (nums[idxs[0]] + nums[idxs[1]] == target)
                {
                    break;
                }
            }
            if (idxs[1] == len) {
                idxs[0]++;
            } else if (nums[idxs[0]] + nums[idxs[1]] == target) {
                break;
            }
        }

        return idxs;
    }
};

int main() {
    std::vector<int> nums = {11, 2, 7, 15, -1};

    Solution sol;
    std::vector<int> idxs = sol.twoSum(nums, 1);

    std::cout << idxs[0] << "," << idxs[1];

    return 0;
}